pkg/promlib/querydata/request.go

```diff
@@ -225,3 +274,15 @@ func (s *QueryData) exemplarQuery(ctx context.Context, c *client.Client, q *mode
 	}()
 	return s.parseResponse(ctx, q, res, enablePrometheusDataplaneFlag)
 }
+
+func addDataResponse(res *backend.DataResponse, dr *backend.DataResponse) {
+	if res.Error != nil {
+		if dr.Error == nil {
+			dr.Error = res.Error
+		} else {
+			dr.Error = fmt.Errorf("%v %w", dr.Error, res.Error)
+		}
+		dr.Status = res.Status
+	}
+	dr.Frames = append(dr.Frames, res.Frames...)
+}
```

Comment:
I am not sure if we need this function as it is used only one place.

Reply:
i updated the code to add the other place where the same code was being used to use the new method instead. kindly check.